libraries/RW/K8s/namespace_tasks_mixin.py

```diff
@@ -235,3 +242,116 @@ def get_event_count(
             return len(event_rows)
         else:
             return 0
+
+    def _convert_age_to_search_time (self, age) -> str: 
+        current_time = datetime.now()
+        age = int(age.split('m')[0])
+        search_time = current_time - timedelta(hours=0, minutes=age) 
+        return search_time.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+
+    def count_events_by_age_and_type (
+        self, 
+        namespace:str,
+        context:str,
+        kubeconfig: platform.Secret,
+        target_service: platform.Service,
+        binary_name: str = "kubectl",
+        event_age: str = None, 
+        event_type: str = "Warning",
+    ) -> float:
+        # K8s Event Ref: https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/event-v1/
+        search_time=self._convert_age_to_search_time(age=event_age)
+        search_filter=f"type==`{event_type}` && lastTimestamp >= `{search_time}`"
+        if "ALL" in namespace: 
```

Comment:
Purely for discussion as this would be a bunch of work for a pattern we might not stick to:

So I've hit this case a handful of times where I check for magic string values like `ALL` depending on what the user passed in. This is a question of cardinality; do we want 1 namespace, a list of namespaces, all, etc, so it would be great if we could just pass in varied types which we could do if we removed the `:str` type cast, however I'd discourage removing types where possible.

Python does have Union'd types so you can do `Union[bool, str, list]` to accept those 3 types for a param, but I recently discovered that robot will cast to only the first type in the union list, causing some interesting casting bugs. This makes me wonder if we want wrapper methods with any/singular types in the wrapper, the wrapper does some type handling to validate, then calls the actual method. This means more work for us so I'd say lets hold off, but food for thought in the long term.